### ISSUE TITLE
Would you be interested in a collaboration on integrating Shortcake with Fronted?

### ISSUE THREAD

**Issue by jesperbjerke** [wp-shortcake/shortcake#501]
I'm the main developer of an open source project called [Fronted](https://github.com/jesperbjerke/wa-fronted), where a 1.0 release is near. After this, I'd like to look into making the Shortcake interface and functionality available on the frontend together with Fronted.

Would you be interested in help doing this? Depending on your interest in this project, I could take on the whole integration part of it, while you assist with answering questions and such.

**Comment by danielbachhuber** [comment]
We could consider integrating Fronted with Shortcake. However, Shortcake is currently closely-coupled with TinyMCE, so I'm not sure it will be an easy task. It might make most sense as a supplementary plugin, with abstractions added to Shortcake as needed.

**Comment by jesperbjerke** [comment]
Yeah, making it as an add-on plugin or something like it was my thought too. At least to a start. I'm sure we'll probably encounter some problems along the way, but at first glance, it looks totally doable. I'll get back to you with a mail and we can discuss this further. If you don't want to have an open discussion here about the integration that is.

**Comment by danielbachhuber** [comment]
> If you don't want to have an open discussion here about the integration that is.

We can just keep this issue open.

**Comment by jesperbjerke** [comment]
Just released v1.1 and Fronted is now utilizing TinyMCE instead of Medium Editor, so this integration should go much more smoothly now. Will start looking into it next :)

**Comment by jesperbjerke** [comment]
Kind of ran into a wall. Having trouble figuring out what I'm missing. Simply added..

``` php
add_action( 'wa_fronted_after_scripts',  array( $this, 'action_admin_enqueue_scripts' ) );
add_action( 'wa_fronted_after_scripts',  array( $this, 'action_wp_enqueue_editor' ) );`
```

..in `private function setup_actions()`. But getting a `Uncaught TypeError: Cannot read property 'replace' of undefined` in `underscore.min.js?ver=1.6.0:5`. Since it's undescore throwing the error, I'm having trouble tracing what is actually causing it. I assume it's some script that's loaded in the admin that I'm missing.

Any insights?

![untitled-1](https://cloud.githubusercontent.com/assets/5323483/10715232/064c012c-7b0f-11e5-82f7-eb5eb198cb62.jpg)

**Comment by danielbachhuber** [comment]
The templates aren't being loaded, because they're added to `admin_print_footer_scripts` https://github.com/fusioneng/Shortcake/blob/master/inc/class-shortcode-ui.php#L240. You'll need to make sure they are rendered on the frontend too.

**Comment by jesperbjerke** [comment]
Just made a pull request adding support for Fronted :)

**Comment by jesperbjerke** [comment]
Ok, so since the pull requests werent accepted. I need a way to call these functions properly:

```
action_admin_enqueue_scripts
action_wp_enqueue_editor
action_admin_print_footer_scripts
```

Maybe if you make them static I can call them like so: `add_action( 'wa_fronted_after_scripts', array( 'Shortcode_UI', 'action_admin_enqueue_scripts' ) );` ?

**Comment by danielbachhuber** [comment]
Because `Shortcode_UI` is a singleton, you can call them like this:

```
add_action( 'wa_fronted_after_scripts', array( Shortcode_UI::get_instance(), 'action_admin_enqueue_scripts' ) );
```

**Comment by jesperbjerke** [comment]
Ah, right, thanks. Next question: could you make the js object `shortcodeViewConstructor` accessible through the global `window.Shortcode_UI` ? Want to access the `.edit()` inside and dont wanna duplicate code.

**Comment by jesperbjerke** [comment]
Edit: noted that it does the same thing in admin. Steps to reproduce: 1. insert shortcode, 2. edit shortcode, 3. press edit again (nothing happens), 4. press edit again (it works)

Here's a weird one: every 2nd time parseShortcodeString gets called it returns undefined.
the `var matches = regexp.exec( shortcodeString );` seems to return null for some reason but sent data is the same, so this is strange.

If you look at the screenshot below, I've called parseShortcodeString 4 times. And as you can see the 1st and 3rd time works fine but not 2nd and 4th. :confounded: 

![sui](https://cloud.githubusercontent.com/assets/5323483/10770078/b47162ca-7ce9-11e5-8cfc-3c633bea86e2.jpg)

**Comment by goldenapples** [comment]
@jesperbjerke Can you check if the issue you were seeing is resolved after #531?

**Comment by jesperbjerke** [comment]
@goldenapples Problem not resolved, sorry. See my comment in the code below

``` javascript
    parseShortcodeString: function( shortcodeString ) {
        var model, attr;

        var shortcode_tags = _.map( sui.shortcodes.pluck( 'shortcode_tag' ), this.pregQuote ).join( '|' );
        var regexp = wp.shortcode.regexp( shortcode_tags );
        var matches = regexp.exec( shortcodeString ); //Returns null every 2nd time it's executed

        if ( ! matches ) {
            return;
        }

        defaultShortcode = sui.shortcodes.findWhere({
            shortcode_tag : matches[2]
        });

        if ( ! defaultShortcode ) {
            return;
        }

        var shortcode = wp.shortcode.fromMatch( matches );
        return this.getShortcodeModel( shortcode );
    },
```
